torch/csrc/jit/passes/concat_opt.cpp

```diff
@@ -497,5 +497,297 @@ void ExpandConcatAndEliminateRedundancy(const std::shared_ptr<Graph>& graph) {
   GRAPH_DUMP("After expanding Concat and eliminating redundancy", graph);
 }
 
+namespace {
+
+class TrivialConcatEliminator {
+ public:
+  explicit TrivialConcatEliminator(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)), aliasDb_(graph_) {}
+
+  bool run() {
+    collectTrivialConcatNodes(graph_->block());
+    auto changed = eliminateTrivialNodes();
+    if (changed) {
+      // EliminateDeadCode to get rid of ListConstructs that we might not need
+      // anymore.
+      EliminateDeadCode(graph_);
+    }
+    return changed;
+  }
+
+ private:
+  // Stores a trivial concat node and the new output to use
+  using TrivialConcat = std::pair<Node*, Value*>;
+
+  bool isTrivialList(const Node* list_node) const {
+    return list_node->kind() == prim::ListConstruct &&
+        list_node->inputs().size() == 1;
+  }
+
+  void handleConcat(Node* node) {
+    auto* list = node->input(0);
+    auto* list_node = list->node();
+    if (isTrivialList(list_node) &&
+        aliasDb_.couldMoveBeforeTopologically(list_node, node)) {
+      auto* new_output = list_node->input(0);
+      trivial_nodes_.emplace_back(node, new_output);
+    }
+  }
+
+  void handleVarConcat(Node* node) {
+    // A node is trivial if it looks like prim::VarConcat(%tensor, %dim)
+    if (node->inputs().size() == 2) {
+      auto* new_output = node->input(0);
+      trivial_nodes_.emplace_back(node, new_output);
+    }
+  }
+
+  void collectTrivialConcatNodes(Block* block) {
+    for (auto* node : block->nodes()) {
+      for (auto* b : node->blocks()) {
+        collectTrivialConcatNodes(b);
+      }
+
+      if (node->kind() == prim::VarConcat) {
+        handleVarConcat(node);
+      } else if (node->kind() == aten::cat) {
+        handleConcat(node);
+      }
+    }
+  }
+
+  bool eliminateTrivialNodes() {
```

Comment:
I decided to abandon the trivial concat stuff. Aside from the aliasing issues, it can cause issues if you don't know the shape of the tensor statically. Specifically, you can have graphs like this:
```
def forward(x):
    return torch.cat([x], 3)
```
which will throw an exception for certain inputs:
```
x = torch.tensor(1.0)
forward(x)
# RuntimeError: zero-dimensional tensor (at position 0) cannot be concatenated
```
Since I haven't seen any models that could benefit from eliminating trivial concats (I implemented it more as a "nice to have" kind of thing), I think it's more trouble than it's worth. 

We conveniently ignore issues like this in a lot of static runtime passes. But since we're hooking this into `PeepholeOptimize`, we need to be more careful.